# Background

## Directory Structure
```
.  
├── scripts  
|   ├── pipeline.sh # include every other bash scripts in order  
|   └── *.sh  
├── data  
|   ├── fastqc  
|   |   ├── *.html  
|   |   └── *.zip  
|   └── trimmomatic  
|       ├── *_paired.fastq.gz  
|       └── *_unpaired.fastq.gz  
├── S288C_reference_genome_R64-2-1_20150113  
|   ├── index  
|   |   └── *.ht2  
|   ├── .gtf of reference genome  
|   ├── .fsa of reference genome before and after cleaning headers  
|   ├── phenotype.csv for ballgown  
|   └── merge information  
├── results  
|   ├── *.sam before merge  
|   ├── *.bam before merge  
|   └── *.gtf before merge  
└── ballgown  
    ├── temp30 (data for temperature 30)  
    └── temp37 (data for temperature 37)  
```

## Environment

### Conda Information
* Miniconda3 version 4.5.12
* Python version 3.7.1.final.0
* Platform: linux-64
  
Configuration File: .miniconda3testrc
```sh
# added by Miniconda3 4.5.12 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/data/users/duanr1/bin/miniconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/data/users/duanr1/bin/miniconda3/etc/profile.d/conda.sh" ]; then
        . "/data/users/duanr1/bin/miniconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/data/users/duanr1/bin/miniconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
```

### Conda Environments
* hisat2
Configuration file [hisat2_env.yml](conda_config/hisat2_env.yml)

* trimmomatic
Configuration file [trimmomatic_env.yml](conda_config/trimmomatic_env.yml)

* r_env
Configuration file [r_env.yml](conda_config/r_env.yml)

***

# Pipeline

## Downloading Data

### Downloading Sample Data
We are using Saccharomyces cerevisiae S228C sample reads from the European Bioinformatics Institute.  
Download all 8 samples by executing ```./sccripts/download_fastq.sh```

### Downloading Reference Data
We are using Saccharomyces cerevisiae S228C Release R64-2-1 from  
https://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/  
Download ```S288C_reference_genome_R64-2-1_20150113.tgz``` and unzip it to ```./S288C_reference_genome_R64-2-1_20150113```

## Preprocessing

#### Quality Measurement of Sample Reads
Use FASTQC to measure the quality scores (phred) across all bases of each sample reads. Results are available at ```./data/fastqc```

#### Cleaning Reference Genome
Execute ```./scripts/clean_fsa.sh``` to remove unnecessary headers of all .fastq files

#### Trimming Sample Reads
Under conda environment trimmomatic, execute ```./scripts/trimmomatic.sh``` to generate both paired reads and unpaired reads at ```./data/trimmomatic```. (Only paired data are used)

#### Building Index
Under conda environment hisat2, execute ```./scripts/index_builder.sh``` to build HISAT2 index for each sample read

#### Converting gff3 to gtf
Under conda environment hisat2, execute ```./scripts/gffread.sh``` to convert reference genome annotation file in .gff3 format to ballgown required format .gtf

## Results

***

# Pipeline Steps

## Description

**1. Map the reads for each sample to the reference genome**  
Use hisat2 to:  
(1) build indices of sample reads  
(2) map sample reads to the reference genome  
(3) output .sam files  
  
**2. Sort and convert the SAM files to BAM**  
Use samtools to sort .sam files to .bam files such that:  
(1) the files can store alignment information generated by various alignment programs  
(2) the files are simple to be generated and converted  
(3) the files are compact in file size  
(4) the files are indexed by genomic position to efficiently retrieve all reads aligning to a locus  
  
**3. Assemble transcripts for each sample**  
Use stringtie to assemble RNA-Seq alignments into potential transcripts and output an estimated .gtf file for each sample  
  
**4. Merge transcripts from all samples**  
Use stringtie to merge all transcripts generated by the previous step into a non-redundant set of transcripts and output a merged .gtf file  
  
**5. Examine how the transcripts compare with the reference annotation (optional)**  
Use gffcompare to compare and compute the estimated accuracy of the merged .gtf file and .gtf files of each sample, to check how the predicted transcripts relate to an annotation file  
  
**6. Estimate transcript abundances and create table counts for Ballgown**  
Use stringtie to generate .gtf file and .bam file for each sample reads based on the merged annotation file

**7. Perform statistical analysis**

* Load relevant R packages

* Load the phenotype data for the samples

* Read in te expression data that were calculated by StringTie.

* Filter to remove low-abundance genes

* Identify transcripts that show statistically significant differences between groups

* Identify genes that show statistically significant differences between groups

* Add gene names and gene IDs to the results_transcripts data frame

* Sort the results from the smallest P value to the largest

* Write the results to a csv file that can be shared and distributed

* Identify transcripts and genes with a q value < 0.5

* Plotting (optional)

* Show the distribution of gene abundances (measured as FPKM values) across samples

* Make plots of individual transcripts across samples

* Plot the structure and expression levels in a sample of all transcripts that share the same gene locus

* Plot the average expression levels for all transcripts of a gene within different groups using the plotMeansfunction

## Code

```./scripts/bash_step1.sh```: Step 1 & 2 & 3  
```./scripts/bash_step2.sh```: Step 4 & 5  
```./scripts/bash_step3.sh```: Step 6  
```./scripts/R_script.sh```: Step 7  
(```./scripts/R_script_pipeline.R``` includes all R codes for step 7)

***

# Conclusion
